commit cf41d221c323666cb358ab71fd92453b99f8f678 (refs/changes/81/2081/3)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2017-06-13T16:12:09-07:00 (2 years, 4 months ago)
    
    OS-6177 avoid checking content-md5 on imgadm import when a checksum exists on image metadata
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/src/img/CHANGES.md b/src/img/CHANGES.md
index 101bafb0..8233633a 100644
--- a/src/img/CHANGES.md
+++ b/src/img/CHANGES.md
@@ -6,6 +6,10 @@ Known issues:
   Docker Registry v1, which is now deprecated.
 
 
+## 3.7.2
+
+- OS-6177 drop md5 hashing from imgadm import
+
 ## 3.7.1
 
 - OS-5823 Allow 'imgadm install' to be able to install a type=docker image.
diff --git a/src/img/lib/imgadm.js b/src/img/lib/imgadm.js
index 8b733c31..64e7c750 100644
--- a/src/img/lib/imgadm.js
+++ b/src/img/lib/imgadm.js
@@ -20,7 +20,7 @@
  *
  * CDDL HEADER END
  *
- * Copyright (c) 2015, Joyent, Inc. All rights reserved.
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
  *
  * * *
  * The main imgadm functionality. The CLI is a light wrapper around this tool.
@@ -1424,10 +1424,12 @@ IMGADM.prototype._downloadImageFile = function _downloadImageFile(opts, cb) {
             if (!zstream && opts.imgMeta.checksum) {
                 ctx.checksum = opts.imgMeta.checksum.split(':');
                 ctx.checksumHash = crypto.createHash(ctx.checksum[0]);
-            }
-            if (ctx.stream.headers['content-md5']) {
+                log.debug({uuid: uuid}, 'creating %s checksum hash',
+                    ctx.checksum[0]);
+            } else if (ctx.stream.headers['content-md5']) {
                 ctx.contentMd5 = ctx.stream.headers['content-md5'];
                 ctx.md5sumHash = crypto.createHash('md5');
+                log.debug({uuid: uuid}, 'creating content-md5 checksum hash');
             }
             ctx.stream.on('data', function (chunk) {
                 if (opts.bar)
diff --git a/src/img/package.json b/src/img/package.json
index 66b86a93..c9e0ba5b 100644
--- a/src/img/package.json
+++ b/src/img/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgadm",
   "description": "Manage SmartOS virtual machine images.",
-  "version": "3.7.1",
+  "version": "3.7.2",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
diff --git a/tools/rsync-to b/tools/rsync-to
index e169470e..a59a8407 100755
--- a/tools/rsync-to
+++ b/tools/rsync-to
@@ -19,17 +19,28 @@ set -o errexit
 TOP=$(cd $(dirname $0)/../; pwd)
 NODE=$1
 
-mounted=$(ssh ${NODE} mount | grep "/var/tmp/vm" || echo "notmounted")
-
 # VM
-if [[ $mounted == "notmounted" ]]; then
+VM_MOUNTED=$(ssh ${NODE} mount | grep "/var/tmp/vm" || echo "notmounted")
+if [[ $VM_MOUNTED == "notmounted" ]]; then
     # Copy the original, rsync and then mount it.
     echo "Creating rw /usr/vm mount"
     ssh ${NODE} cp -RP /usr/vm /var/tmp/vm
     ssh ${NODE} mount -F lofs /var/tmp/vm /usr/vm
 fi
-
 echo "Updating /usr/vm/node_modules"
 rsync -av ${TOP}/src/vm/node_modules \
     $NODE:/usr/vm/ \
     --exclude .git
+
+# IMG
+IMG_MOUNTED=$(ssh ${NODE} mount | grep "/var/tmp/img" || echo "notmounted")
+if [[ $IMG_MOUNTED == "notmounted" ]]; then
+    # Copy the original, rsync and then mount it.
+    echo "Creating rw /usr/img mount"
+    ssh ${NODE} cp -RP /usr/img /var/tmp/img
+    ssh ${NODE} mount -F lofs /var/tmp/img /usr/img
+fi
+echo "Updating /usr/img"
+rsync -av ${TOP}/src/img \
+    $NODE:/usr/ \
+    --exclude .git
